source/simulator/melt.cc

```diff
@@ -255,23 +263,19 @@ namespace aspect
                                              pressure_scaling *
                                              pressure_scaling *
                                              (scratch.phi_p_c[i] * scratch.phi_p_c[j])
+                                             +
+                                             // add S between p_c and p_f
+                                             (p_c_scale * one_over_eta *
+                                              pressure_scaling *
+                                              pressure_scaling)
+                                             * scratch.phi_p[i] * scratch.phi_p_c[j]
+                                             +
+                                             (p_c_scale * one_over_eta *
+                                              pressure_scaling *
+                                              pressure_scaling)
+                                             * scratch.phi_p_c[i] * scratch.phi_p[j]
                                             )
                                             * JxW;
-                // add S between p_c and p_f
-                if (true)
-                  data.local_matrix(i,j) +=
-                    (
-                      (p_c_scale * one_over_eta *
-                       pressure_scaling *
-                       pressure_scaling)
-                      * scratch.phi_p[i] * scratch.phi_p_c[j]
-                      +
-                      (p_c_scale * one_over_eta *
-                       pressure_scaling *
-                       pressure_scaling)
-                      * scratch.phi_p_c[i] * scratch.phi_p[j]
-                    )
-                    * JxW;
```

Comment:
The above lines might be the cause for the test changes. Previously you always added these line (`if(true)`, although that condition is unnecessary), now you only add them if the condition above is true (component_index == component_index). Revert to the old state, but remove the `if(true)`. Its only the preconditioner though, not sure this is the only reason.

Reply:
Ah, I see, I'll try it. Thanks!